lib/screens/translation_screen.dart

```diff
@@ -0,0 +1,360 @@
+
+    _navigateToNextSentencePage();
+  }
+
+  void _cancelRecording({BuildContext context}) {
+    String fileId = _currentSentence.id;
+    final recordedFile = File("$recordStorage/$fileId.$fileExtension");
+
+    recordedFile.delete().then((result) {
+      setState(() {
+        _recorderTimer = "";
+        _isRecording = false;
+      });
+
+      _showMessage(deletionSuccess, context: context);
+    }).catchError((err) {
+      _showMessage(deletionError, context: context);
+    });
+  }
+
+  void _showMessage(String msg, {BuildContext context}) {
+    Flushbar(
+        message: msg,
+        duration: Duration(seconds: 3),
+        margin: EdgeInsets.all(8),
+        borderRadius: 8,
+        flushbarStyle: FlushbarStyle.FLOATING,
+        icon: Icon(
+          Icons.info_outline,
+          size: 28.0,
+          color: Theme.of(context).primaryColorDark,
+        ),
+        leftBarIndicatorColor: Theme.of(context).primaryColorDark)
+      ..show(context);
+  }
+
+  void _askToSaveRecording() {
+    showDialog<void>(
+        context: context,
+        barrierDismissible: false,
+        builder: (BuildContext context) {
+          return AlertDialog(
+              titleTextStyle: TextStyle(
+                  color: Theme.of(context).primaryColorDark,
+                  fontWeight: FontWeight.bold,
+                  fontSize: 20.0),
+              title: Text("Save Recording"),
+              content: Text("Are you satisfied with your recording?"),
+              actions: <Widget>[
+                FlatButton(
+                  color: _isRecordingColor["primaryDark"],
+                  textColor: Colors.white,
+                  child: Text('No'),
+                  onPressed: () {
+                    _cancelRecording(context: context);
+                    Navigator.of(context).pop();
+                  },
+                ),
+                FlatButton(
+                  color: Theme.of(context).primaryColorDark,
+                  textColor: Colors.white,
+                  child: Text('Yes'),
+                  onPressed: () {
+                    _doneRecording();
+                    Navigator.of(context).pop();
+                  },
+                ),
+              ]);
+        });
+  }
+
+  void _navigateToNextSentencePage() {
+    if (_pageController.hasClients) {
+      _pageController.animateToPage(
+        _progression > 0 ? _progression : 1,
+        duration: const Duration(milliseconds: 800),
+        curve: Curves.bounceInOut,
+      );
+    }
+  }
+
+  Widget _recorderWidget() {
+    Widget stopIcon = Icon(Icons.stop,
+        size: _recorderIconSize, color: _isRecordingColor["accent"]);
+    Widget micIcon = Icon(Icons.mic,
+        size: _recorderIconSize, color: Theme.of(context).accentColor);
+
+    Color darkColor = Theme.of(context).primaryColorDark;
+    Color lightColor = Theme.of(context).primaryColorLight;
+    if (_isRecording) {
+      darkColor = _isRecordingColor["primaryDark"];
```

Comment:
I whish there was a way not to use magic strings for "primaryDark", "accent",...